kotlinx-coroutines-core/jvm/src/internal/FastServiceLoader.kt

```diff
@@ -5,6 +5,13 @@ import java.net.*
 import java.util.*
 import java.util.jar.*
 import java.util.zip.*
+import kotlin.collections.ArrayList
+
+/**
+ * TODO investigate possible java.vendor values for fallbacks
+ * Don't use JvmField here to enable R8 optimizations via "assumenosideeffects"
+ */
+internal val ANDROID_DETECTED = systemProp("java.vendor")?.toLowerCase()?.contains("android") ?: false
```

Comment:
I've seen this used in the past, but I'm not sure anything actually guarantees that this string is present. Another reliable way to check is to try to load any Android class (e.g., `Class.forName("android.os.Build")` and then set true or false. R8 will eliminate the lookup so it will become a constant value which should then be inlined so the `if` and thus `load` are removed.